AWSAuthSDK/Sources/AWSMobileClient/AWSMobileClient.swift

```diff
@@ -459,13 +462,26 @@ final public class AWSMobileClient: _AWSMobileClient {
                     signInInfo[self.TokenKey] = session.accessToken!.tokenString
                     signInInfo[self.ProviderKey] = "OAuth"
                     
-                    self.performHostedUISuccessfulSignInTasks(disableFederation: hostedUIOptions.disableFederation, session: session, federationToken: federationToken!, federationProviderIdentifier: federationProviderIdentifier, signInInfo: &signInInfo)
-                    completionHandler(.signedIn, nil)
-                    if self.pendingGetTokensCompletion != nil {
-                        self.tokenFetchLock.leave()
+                    self.performHostedUISuccessfulSignInTasks(disableFederation: hostedUIOptions.disableFederation,
+                                                              session: session,
+                                                              federationToken: federationToken!,
+                                                              federationProviderIdentifier: federationProviderIdentifier,
+                                                              signInInfo: &signInInfo).continueWith() { task in
+                                                                
+                                                                if let error = task.error {
+                                                                    completionHandler(nil, error)
+                                                                } else {
+                                                                    self.mobileClientStatusChanged(userState: .signedIn,
+                                                                                                   additionalInfo: signInInfo)
+                                                                    completionHandler(.signedIn, nil)
+                                                                    if self.pendingGetTokensCompletion != nil {
+                                                                        self.tokenFetchLock.leave()
+                                                                    }
+                                                                    self.pendingGetTokensCompletion?(self.getTokensForCognitoAuthSession(session: session), nil)
```

Comment:
It's a bit hard for me to follow--is it safe that `pendingGetTokensCompletion` is being invoked after the completion handlers & state listener?

Reply:
Moved this above completion and state listener. So the developer will get informed in the following order:
1. `pendingGetTokensCompletion` (if it was set)
2. state listener
3. completion handler